src/internal/observable/from.ts

```diff
@@ -12,6 +12,90 @@ import { ObservableInput, SchedulerLike } from '../types';
 
 export function from<T>(input: ObservableInput<T>, scheduler?: SchedulerLike): Observable<T>;
 export function from<T>(input: ObservableInput<ObservableInput<T>>, scheduler?: SchedulerLike): Observable<Observable<T>>;
+
+/**
+ * Creates an Observable from an Array, an array-like object, a Promise, an iterable object, or an Observable-like object.
+ *
+ * <span class="informal">Converts almost anything to an Observable.</span>
+ *
+ * ![](from.png)
+ *
+ * `from` converts various other objects and data types into Observables. It also converts a Promise, an array-like, or an
+ * <a href="https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#iterable" target="_blank">iterable</a>
+ * object into an Observable that emits the items in that promise, array, or iterable. A String, in this context, is treated
+ * as an array of characters. Observable-like objects (contains a function named with the ES2015 Symbol for Observable) can also be
+ * converted through this operator.
+ *
+ * ## Examples
+ * ### Converts an array to an Observable
+ * ```javascript
+ * import { from } from 'rxjs/observable/from';
+ *
+ * const array = [10, 20, 30];
+ * const result = from(array);
+ *
+ * result.subscribe(x => console.log(x));
+ *
+ * // Logs:
+ * // 10 20 30
+ * ```
+ *
+ * ---
+ *
+ * ### Convert an infinite iterable (from a generator) to an Observable
+ * ```javascript
+ * import { take } from 'rxjs/operators';
+ * import { from } from 'rxjs/observable/from';
+ *
+ * function* generateDoubles(seed) {
+ *    let i = seed;
+ *    while (true) {
+ *      yield i;
+ *      i = 2 * i; // double it
+ *    }
+ * }
+ *
+ * const iterator = generateDoubles(3);
+ * const result = from(iterator).pipe(take(10));
+ *
+ * result.subscribe(x => console.log(x));
+ *
+ * // Logs:
+ * // 3 6 12 24 48 96 192 384 768 1536
+ * ```
+ *
+ * ---
+ *
+ * ### with async scheduler
+ * ```javascript
+ * import { from } from 'rxjs/observable/from';
+ * import { async } from 'rxjs/scheduler/async';
+ *
+ * console.log('start');
+ *
+ * const array = [10, 20, 30];
+ * const result = from(array, async);
+ *
+ * result.subscribe(x => console.log(x));
+ *
+ * console.log('end');
+ *
+ * // Logs:
+ * // start end 10 20 30
+ * ```
+ *
+ * @see {@link fromEvent}
+ * @see {@link fromEventPattern}
+ * @see {@link fromPromise}
+ *
+ * @param {ArrayLike<T> | ObservableInput<T>} A subscription object, a Promise, an Observable-like,
```

Comment:
ObserableInput is a type of `SubscribableOrPromise<T> | ArrayLike<T> | Iterable<T>` therefore adding `ArrayLike<T>` isn't really needed